### ISSUE TITLE
Add warning when featured image does not meet Google Search requirements

### ISSUE THREAD

**Issue by westonruter** [ampproject/amp-wp#1812]
For Schema.org metadata, the featured image of a post is used as the `image` property. For this property, [Google Search requires](https://developers.google.com/search/docs/data-types/article#article_types):

> The URL to an image that is representative of the article or AMP story.
> 
> Due to format differences in search results, the following image guidelines only apply to general AMP pages, not AMP stories. AMP stories have different requirements for images.
> 
> * Only a marked-up image that directly belongs to the article should be specified.
Images should be at least 1200 pixels wide.
> * Every page must contain at least one image (whether or not you include markup). Google will pick the best image to display in Search results based on the aspect ratio and resolution.
> * Image URLs must be crawlable and indexable.
> * Images must represent the marked up content.
> * Images must be in .jpg, .png, or .gif format.
> * For best results, provide multiple high-resolution images (minimum of 800,000 pixels when multiplying width and height) with the following aspect ratios: 16x9, 4x3, and 1x1.

There is currently no guidance offered by the plugin for the featured image that is selected for a post. There should be a warning when the image is not 1200 pixels wide.

If the requirements are not met, should the image be excluded altogether? Is it better to have something rather than nothing?

We need to avoid situations where Google Search Console informs users of problems related to pages that the AMP plugin generates.

See also [plugin forum topic](https://wordpress.org/support/topic/google-search-console-image-size-smaller-than-recommended-size/).

**Comment by westonruter** [comment]
Also, per:

> For best results, provide multiple high-resolution images (minimum of 800,000 pixels when multiplying width and height) with the following aspect ratios: 16x9, 4x3, and 1x1.

We should loop over all registered image sizes to find if there are others with these orientations, and then include all images that are at least 1200px wide, although it is unlikely that there will be any at that size.

**Comment by robin-scott** [comment]
Confirming this is dropping onto a lot of backlogs right now - and I've been looking at a couple of solutions, one of which is a default image which complies with the specifications which could be substituted, although this "Only a marked-up image that directly belongs to the article should be specified." does appear to preclude that :|

This is "just" a warning, and Google does make a marked distinction between warning and error when it comes to indexing and Schema in my experience.

**Comment by swissspidy** [comment]
> It should probably also warn when no featured image is selected at all, since this is also a GSC warning.

Just had this situation on a friend's site yesterday. No changes to the site, but GSC added hundreds of warnings because there was no featured image or anything. On these pages, the only image is the site icon (which of course isn't 1200px wide)

**Comment by robin-scott** [comment]
Just popping back to mention that the resolution which worked across a few clients (not a fix for this plugin issue generally, but perhaps helpful if some WP site owner hops on here) was to replace the featured image for a 1200px+ wide pic. This removes the Search Console errors which began displaying this week.

It's written up here: https://robinjescott.com/new-google-amp-warning-image-size-smaller-than-recommended-size/

I'm actively working on a couple of more technical solutions for larger sites where this has presented in a less easy to fix way (read thousands of warnings!). If any are worth PR here I will do so.

**Comment by westonruter** [comment]
@robin-scott Just in case someone mistakes the action item from your post to be to take the above 1200px AMP logo image to be used as the featured image: please don't do this. Using a generic 1200px image would violate these requirements: 

> Only a marked-up image that directly belongs to the article should be specified.
and
> Images must represent the marked up content.”

While GSC will have difficulty validating this requirement since it requires some artificial intelligence, but I think it would a good bet to assume that Google will introduce the AI to make sure the image matches the content in the future. At that point, using such a generic image would again be flagged as a warning.

**Comment by westonruter** [comment]
A possible automated approach could involve either upscaling a smaller image to be 1200px wide, which would normally look fine since the image is normally displayed as a thumbnail anyway in search results. Alternatively, the image could be made 1200px wide by use of letterboxing with either solid color margins or using the blurring the original image in the background, as is often seen when a portrait video is shown inside of a landscape video.

**Comment by robin-scott** [comment]
Yeah agree - don't do a generic image.  I am leaning towards letterboxing as upscaling can cause resolution issues particularly if the non amp site featured image was particularly small.

**Comment by luisfranco** [comment]
I have checked, and images not only need to be 1200px wide, but these must also have any of the following aspect ratios: 16x9, 4x3, and 1x1.

Something like this:
```"image": {
        "@type": "ImageObject",
        "height": 1200,
        "url": "https://domain/images/filename.png",
        "width": 360
    }
```

Will trigger the same `image too small` error, it only validated on GSC until I used an image with the following dimensions 1200x675 (16x9). 

I was able to pass this validation with an image without the required dimensions by using:

```
"image": [
  "https://domain/images/filename.png"
]
```

But probably Google will be able to find out the image doesn't meet their requirements.
